### PR TITLE
Add Ruby 3.0–3.2 support, drop Ruby 2.6

### PR DESCRIPTION
**Pull request by razumau**
### Description
Adds tests on Ruby 3.0, 3.1, 3.2. Drops support for Ruby 2.6.

To enable Ruby 3.2 support, I updated Nokogiri to the most recent version. That led to two changes:
1. Invalid HTML with unpaired opening brackets is now truncated to a non-HTML string, e.g., `"<<p>hello there</p>"` will be truncated to `"&lt;<p></p>"`. Previously, it was truncated to a valid HTML string `"<p>&hellip;</p>"`. This is caused by [a change in libxml2](https://gitlab.gnome.org/GNOME/libxml2/-/issues/339), which prevents potential quadratic behaviour and DoS attacks. Users relying on abbreviato’s output being a valid HTML string should now check input’s validity first.
2. In tests, `:strict` mode for HTML validation has been dropped. [A change in Nokogiri 1.11](https://nokogiri.org/CHANGELOG.html#html-parser-now-obeys-the-strict-or-norecover-parsing-option) made the strict mode stricter, and dropping that flag is the default recommendation in the Nokogiri release notes.

I’ve also cleaned up workflows, moved development dependencies into Gemfile, and added a changelog.

### CODE REVIEW

**Review by HeyNonster — DISMISSED** [review]
@mariozaizar locked Nokogiri to `1.10.10` in #55 (because of the failing specs, which are fixed now).

@mariozaizar do we need to give any teams a heads up about the Nokogiri breaking changes?

@razumau, we can probably remove `bump` 🤷

**Comment by razumau** [comment]
> do we need to give any teams a heads up about the Nokogiri breaking changes?

I’ve made a post about this on ask-email-eng.

File: CONTRIBUTING.md
Comment (razumau) [inline CONTRIBUTING.md:17]: I’ll rewrite this section in the next PR, with the publishing workflow onboarding.

**Comment by mariozaizar** [comment]
> @mariozaizar do we need to give any teams a heads up about the Nokogiri breaking changes?

@HeyNonster, I only know about Email, but bumping this with a major tag to ensure that breaking changes are expected makes sense 🤔 . I saw that you folks added a Changelog, that also helps 🎖️

**Review by mariozaizar — APPROVED** [review]
LGTM

**Comment by mariozaizar** [comment]
Hey @razumau are you ok if we get this merged? Do you remember if there was anything else preventing the merge? Were you waiting for more reviews?

**Review by HeyNonster — APPROVED** [review]

**Comment by razumau** [comment]
> Hey @razumau are you ok if we get this merged? Do you remember if there was anything else preventing the merge? Were you waiting for more reviews?

Actually, I thought that you wanted to get a review from strongbad. Will merge it now.

**Pull request merged**
